src/aws/mod.rs

```diff
@@ -94,6 +96,24 @@ fn check_response(res: &HttpResponse) -> bool {
         || (status.is_client_error() && re.is_match(response_body.as_ref()))
 }
 
+/// Creates the http connector that has been configured to use the given proxy and TLS settings.
+/// All AWS requests should use this connector as the aws crates by default use RustTLS which we
+/// have turned off as we want to consistently use openssl.
+fn connector(
+    proxy: &ProxyConfig,
+    tls_options: &Option<TlsConfig>,
+) -> crate::Result<SharedHttpClient> {
```

Comment:
Will this function ever be useful from outside of Vector, or is it effectively "exported" by think like the client builder that is already `pub`?

Reply:
It's only really useful for use with the AWS crates. Would this be useful it OPW? I can make it `pub` if so.

Reply:
That was what I was wondering. I don't _know_ we'll need it, so just leave it for now.